plugins/org.jkiss.dbeaver.ext.clickhouse/src/org/jkiss/dbeaver/ext/clickhouse/model/ClickhouseTableColumn.java

```diff
@@ -43,6 +43,10 @@ public String getFullTypeName() {
     @Override
     public void setFullTypeName(String fullTypeName) throws DBException {
         this.fullTypeName = fullTypeName;
+        if (fullTypeName.toUpperCase().contains("ARRAY")) {
+            setTypeName(fullTypeName);
+            return;
+        }
         super.setFullTypeName(fullTypeName);
```

Comment:
I don't see the reason to use `return` here when you have `else`.
```suggestion
        if (fullTypeName.toUpperCase().contains("ARRAY")) {
            setTypeName(fullTypeName);
        } else {
            super.setFullTypeName(fullTypeName);
        }
```

Reply:
Fixed